woocommerce/woocommerce-ios
# Order Creation: Align text fields in the address form

Closes: https://github.com/woocommerce/woocommerce-ios/issues/6018.

## Description

This PR updates Address Form to align textfields vertically. This also fixes issue with missing trailing spaces input.

## Changes
- Adds `MaxWidthPreferenceKey` to a title view in `TitleAndTextFieldRow`.
- Updates `SingleAddressForm` to store shared (max) title width and inject it back into all fields.

### Known limitations
- `TitleAndValueRow` (Country and City fields) wasn't updated yet, but it doesn't have "missing spaces" bug and can be updated in a same way after testing and approval of this PR.

## Testing

1. Open Address Form from New Order or Order Details flows.
2. Enter some data in different fields and confirm that alignment is correct.
3. Update dynamic type size and recheck the layout.

## Screenshots

before|after
--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/163026314-5e7783ab-094a-4b5e-b720-ce342853645f.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/163026350-90a4c09c-83d5-4bfc-ad0b-adf9711531e4.png)
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/163026442-073442a4-c6d4-4c56-9726-606d796677ee.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/163026467-5d32f40c-9bb5-4f61-9d96-98a9f5597488.png)


---
